### DISCUSSION TITLE
Implication operator for IF statements with return.

### DISCUSSION THREAD

**Discussion by janseris** [dotnet/csharplang#6415] [Language Ideas]
Often there are `if` statements without `else` branch because they contain a `return` statement.
My proposal is to use a new `=>` (implication) operator for these cases.

Consider the following scenario:
```
if(property is null)
{
    return null;
}
```

Could be rewritten as:
`(property is null) => null;`

This would be similar to existing inline properties syntax:

Non-inline example: 
`public property Name { get { return _name; } }`

Inline property example:
`public property Name => _name;`

**Comment by HaloFour** [comment]
The `=>` operator isn't an implied return operator.  For a member it indicates that the body of that member is a single expression, but within an expression it indicates the expression that it evaluates to.  This is already how `switch` expressions work today.
